packages/http-client/src/http-client.ts

```diff
@@ -487,6 +496,15 @@ export function getAxiosConfigWithDefaultsWithoutMethod(): Omit<
   };
 }
 
+/**
+ * @internal
+ */
+export function getDefaultHttpRequestConfigOptions(): HttpRequestConfigWithOrigin {
+  return {
+    method: 'get'
```

Comment:
[q] Could one perhaps add `url:'/'` here so that no if statement is needed in the csrf-token-header?

Reply:
I think to make it 100% correct we would have to make a conditional type for the RequestConfig: If the method is post the URL is mandatory. If the method is not POST the URL is optional in the config.